### PR TITLE
fix: resolve invalid link for `profile`

### PR DESCRIPTION
**Pull request by tenshiAMD**
**What**:
- Fixes https://github.com/all-contributors/app/issues/420

**Why**:
- Some profile link are invalid due to missing protocol

**How**:
- Check and correct links by adding protocol

**Checklist**:

- [x] Documentation
- [x] Ready to be merged 
- [x] Added myself to contributors table. 
[Bot Usage](https://allcontributors.org/docs/en/bot/installation#4-update-your-contributing-documentation)

### CODE REVIEW

**Comment by tenshiAMD** [comment]
@all-contributors/core gentle ping

File: test/unit/helpers.test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { getProfileLink } = require("../../lib/modules/helpers");
+
+describe("getProfileLink", () => {
+  test("return valid link", async () => {
```
Comment (Roshanjossey) [inline test/unit/helpers.test.js:4]: It would be good to write test cases for other cases as well. In test description, write test cases. Also think about edge cases.

File: lib/modules/helpers.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+function getProfileLink(url) {
+  if (!url.startsWith("http")) return `http://${url}`
```
Comment (Roshanjossey) [inline lib/modules/helpers.js:2]: Better to have `https` than `http`

**Review by Roshanjossey — CHANGES_REQUESTED** [review]

File: lib/modules/helpers.js (hunk 2 above)
Comment (tenshiAMD) [reply lib/modules/helpers.js:2]: @Roshanjossey we can simply do that but not all links setup to work on `https`

File: test/unit/helpers.test.js (hunk 1 above)
Comment (tenshiAMD) [reply test/unit/helpers.test.js:4]: > Also think about edge cases.

@Roshanjossey can you elaborate this more?

File: lib/modules/helpers.js (hunk 2 above)
Comment (Roshanjossey) [reply lib/modules/helpers.js:2]: With browsers like Google Chrome and Brave, it'll automatically switch to `https` but in my experience, Firefox or Chromium doesn't do that.

File: test/unit/helpers.test.js (hunk 1 above)
Comment (Roshanjossey) [reply test/unit/helpers.test.js:4]: Yeah, something like `getProfileLink("httpsamurai.me")`

Also, this `blog` field is a simple text filed for GitHub. I don't see them doing a url validation there.
Right now, I've set mine as the word `contributor` https://api.github.com/users/roshanjossey
Although we expect users to put a valid URL there, it doesn't have to be. Those would also be edge cases.

Comment (tenshiAMD) [reply test/unit/helpers.test.js]: @Roshanjossey done kindly check

File: lib/modules/helpers.js (hunk 2 above)
Comment (tenshiAMD) [reply lib/modules/helpers.js]: @Roshanjossey still some site only use `http` though they need to setup redirect so using `http` will support all

**Comment by tenshiAMD** [comment]
@all-contributors/core gentle ping

**Review by gr2m — APPROVED** [review]

File: test/unit/helpers.test.js (hunk 1 above)
Comment (Roshanjossey) [reply test/unit/helpers.test.js]: Hi @tenshiAMD, checked again.

```js
expect(generateValidLink("httpsamurai.me")).toEqual(`http://httpsamurai.me`);
```
will fail. [TS playground link. Click Run to see result](https://www.typescriptlang.org/pt/play?#code/GYVwdgxgLglg9mABAcwKZlQJwIZVQNWwBsYATAGRjAGsAKETIgLkQGcpMrkAaREVrGGwBbVIgC8iAORSAlIgDeAKESIiqKIgBuxMpRoS+jFYggJ223aQBKqNAA9DAegB6ACyhQADqwD8TAB0nINp-AHcIgIA6WV8AbQBabASALwBBBIAtAAYEgE4AASYAUiiAfQCAagA-AGJxAF0FAEZuACYAVgA2AF9ouOT0rNy82lkm1t6AgCNQpkTBjJz8saLiisqout8AMiDGgCpZABInJRMYYERaAEIdEgoqaij2bEwoVgB1GCg3WgAiDzef6yeT3PRPQwAAyBXiYTicxwUDCIPScUIuV1u4MeNCiwlwED+ONsDlBlge+mo0NhrHhTmQPzcIGmUTMwkRyIEmCEojRGJMmA0DCQOKpSh65zMYFYcHUUSIcGQtDQGBweEIlKeANpIgY2Bg+NQIIk4kkgM8cIRuuE+sNohBUvMctQCqVKvQWFwBCsVJ1ltYepw9uNoKAA)

Again, these are edge cases. You don't have to worry about these if you're happy with the current changes. If not, I could do some refactoring.

Comment (tenshiAMD) [reply test/unit/helpers.test.js]: > If not, I could do some refactoring.

@Roshanjossey sure go ahead. PRs are welcome.

Comment (Roshanjossey) [reply test/unit/helpers.test.js]: Cool cool cool. I'll try next week.
Were my comments useful? Do you need help with reviews?

Comment (tenshiAMD) [reply test/unit/helpers.test.js]: @Roshanjossey done I already tweak it. See https://github.com/all-contributors/app/pull/422. Thanks! 🎉
